aten/src/ATen/native/native_functions.yaml

```diff
@@ -2119,7 +2119,7 @@
 - func: type_as(Tensor self, Tensor other) -> Tensor
   variants: method
 
-- func: _has_same_tensorimpl_type(Tensor self, Tensor other) -> bool
+- func: _has_compatible_type_for_copy(Tensor self, Tensor other) -> bool
```

Comment:
1) I know you already prefixed this with "_", but this name makes it sound like it's (data) copy related, when it's really shallow copy related.  I'd just name this to something scarier, like `_has_compatible_shallow_copy_type` or something (just something mentioning either shallow copy or tensorimpl is sufficient).
2) Can you rename `other` to `from` or `to`?  I know it doesn't matter in the implementation for now, but I know if I see the name "same type" (as before) it doesn't matter what order I pass the parameters, but if I see the "copy" I immediately look what is "from" or "to" and if it's not obvious from the names (like this), I have to look up the implementation.